packages/hierarchy-builder/src/test/internal/operators/Grouping.test.ts

```diff
@@ -0,0 +1,207 @@
+
+import { expect } from "chai";
+import { from } from "rxjs";
+import sinon from "sinon";
+import { LogLevel } from "@itwin/core-bentley";
+import { HierarchyNode } from "../../../hierarchy-builder/HierarchyNode";
+import {
+  createGroupingHandlers,
+  createGroupingOperator,
+  GroupingHandler,
+  GroupingHandlerResult,
+  LOGGING_NAMESPACE,
+} from "../../../hierarchy-builder/internal/operators/Grouping";
+import * as baseClassGrouping from "../../../hierarchy-builder/internal/operators/grouping/BaseClassGrouping";
+import * as classGrouping from "../../../hierarchy-builder/internal/operators/grouping/ClassGrouping";
+import * as groupHiding from "../../../hierarchy-builder/internal/operators/grouping/GroupHiding";
+import * as labelGrouping from "../../../hierarchy-builder/internal/operators/grouping/LabelGrouping";
+import { IMetadataProvider } from "../../../hierarchy-builder/Metadata";
+import { createTestNode, getObservableResult, setupLogging } from "../../Utils";
+
+describe("Grouping", () => {
+  const metadata = {} as unknown as IMetadataProvider;
+
+  before(() => {
+    setupLogging([{ namespace: LOGGING_NAMESPACE, level: LogLevel.Trace }]);
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  describe("createGroupingOperator", () => {
+    let applyGroupingHidingParamsStub: sinon.SinonStub;
+    beforeEach(() => {
+      applyGroupingHidingParamsStub = sinon.stub(groupHiding, "applyGroupHidingParams").callsFake((props) => props);
+    });
+
+    it("doesn't change input nodes when grouping handlers don't group", async () => {
+      const nodes = [
+        createTestNode({
+          key: { type: "instances", instanceKeys: [{ className: "TestSchema.A", id: "0x1" }] },
+          label: "1",
+          params: {
+            grouping: {
+              byBaseClasses: {
+                fullClassNames: ["TestSchema.TestParentClassA", "TestSchema.TestParentClassAA"],
+              },
+              byClass: true,
+              byLabel: true,
+            },
+          },
+        }),
+        createTestNode({
+          key: { type: "instances", instanceKeys: [{ className: "TestSchema.A", id: "0x2" }] },
+          label: "1",
+          params: {
+            grouping: {
+              byBaseClasses: {
+                fullClassNames: ["TestSchema.TestParentClassA", "TestSchema.TestParentClassAA"],
+                hideIfNoSiblings: true,
+              },
+              byLabel: true,
+              byClass: true,
+            },
+          },
+        }),
+      ];
+
+      const result = await getObservableResult(
+        from(nodes).pipe(
+          createGroupingOperator(metadata, [
+            async (allNodes) => {
+              return { grouped: [], ungrouped: allNodes, groupingType: "label" } as GroupingHandlerResult;
+            },
+            async (allNodes) => {
+              return { grouped: [], ungrouped: allNodes, groupingType: "class" } as GroupingHandlerResult;
+            },
```

Comment:
Could this be simplified to the following?
```suggestion
            async (allNodes) => ({ grouped: [], ungrouped: allNodes, groupingType: "label" }),
            async (allNodes) => ({ grouped: [], ungrouped: allNodes, groupingType: "class" }),
```